syweb/webclient/app/components/matrix/notification-service.js

```diff
@@ -48,23 +49,293 @@ angular.module('notificationService', ['matrixService'])
+                    'pattern': tprule.rule_id
+                });
+                break;
+            case 'sender':
+                if (!tprule.rule_id) return null;
+                rawrule.conditions.push({
+                    'kind': 'event_match',
+                    'key': 'user_id',
+                    'pattern': tprule.rule_id
+                });
+                break;
+            case 'content':
+                if (!tprule.pattern) return null;
+                rawrule.conditions.push({
+                    'kind': 'event_match',
+                    'key': 'content.body',
+                    'pattern': tprule.pattern
+                });
+                break;
+        }
+        if (device) {
+            rawrule.conditions.push({
+                'kind': 'device',
+                'profile_tag': device,
+            });
+        }
+        return rawrule;
+    };
+
+    var ruleMatchesEvent = function(rule, ev) {
+        var ret = true;
+        angular.forEach(rule.conditions, function(cond) {
+            ret &= eventFulfillsCondition(cond, ev);
+        });
+        //console.log("Rule "+rule.rule_id+(ret ? " matches" : " doesn't match"));
+        return ret;
+    };
+
+    var eventFulfillsCondition = function(cond, ev) {
+        var condition_functions = {
+            "event_match": eventFulfillsEventMatchCondition,
+            "device": eventFulfillsDeviceCondition,
+            "contains_display_name": eventFulfillsDisplayNameCondition,
+            "room_member_count": eventFulfillsRoomMemberCountCondition
+        };
+        if (condition_functions[cond.kind]) return condition_functions[cond.kind](cond, ev);
+        return true;
+    };
+
+    var eventFulfillsRoomMemberCountCondition = function(cond, ev) {
+        if (!cond.is) return false;
+
+        var room = modelService.getKnownRoom(ev.room_id);
+        if (!room || !room.current_room_state || !room.current_room_state.members) return false;
+
+        var memberCount = Object.keys(room.current_room_state.members).length;
+
+        var m = cond.is.match(/^([=<>]*)([0-9]*)$/);
+        if (!m) return false;
+        var ineq = m[1];
+        var rhs = parseInt(m[2]);
+        if (isNaN(rhs)) return false;
+        switch (ineq) {
+            case '':
+            case '==':
+                return memberCount == rhs;
+            case '<':
+                return memberCount < rhs;
+            case '>':
+                return memberCount > rhs;
+            case '<=':
+                return memberCount <= rhs;
+            case '>=':
+                return memberCount >= rhs;
+            default:
+                return false;
+        }
+    };
+
+    var eventFulfillsDisplayNameCondition = function(cond, ev) {
+        if (!ev.content || ! ev.content.body || typeof ev.content.body != 'string') return false;
+
+        var displayname = $filter("mUserDisplayName")(matrixService.config().user_id, ev.room_id);
```

Comment:
Nitpick: I would prefer if the filter was explicitly set as a dependency to this service, rather than pulling in a dependency on the entire ``$filter`` space. You can do this by specifying a dependency with the name of the filter with the suffix ``Filter`` e.g. ``mUserDisplayNameFilter``, see ``app/room/room-controller.js:19`` for a working example.